tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -2095,3 +2095,293 @@ def call(self, inputs, state):
+        [self._num_units, self._num_units, self._num_units, self._num_units],
+        True, bias_start=self._bias_start, alpha_start=self._alpha_start,
+        beta_start=self._beta_start)
+
+    # i = input_gate, j = new_input, f = forget_gate, o = output_gate
+    i, j, f, o = array_ops.split(value=concat, num_or_size_splits=4, axis=1)
+
+    new_c = (c * math_ops.sigmoid(f + self._forget_bias)
+                + math_ops.sigmoid(i) * self._activation(j))
+    new_h = self._activation(new_c) * math_ops.sigmoid(o)
+
+    if self._state_is_tuple:
+      new_state = rnn_cell_impl.LSTMStateTuple(new_c, new_h)
+    else:
+      new_state = array_ops.concat([new_c, new_h], 1)
+    return new_h, new_state
+
+
+def _multiplicative_integration(args, output_sizes, bias, bias_start=0.0, 
+    alpha_start=1.0, beta_start=1.0):
+  """Multiplicative Integration: alpha * args[0] * W[0] + beta1 * args[1] * W[1] 
+            + beta1 * args[2] * W[2],
+        where alpha, beta1, beta2 and W[i] are variables.
+
+  Args:
+    args: a list of 2D, batch x n, Tensors.
+    output_sizes: a list of second dimension of W[i], where list[i] is int.
+    bias: boolean, whether to add a bias term or not.
+    bias_start: float, starting value to initialize the bias; 0 by default.
+    alpha_start: float, starting value to initialize the alpha; 
+      1 by default.
+    beta_start: float, starting value to initialize the beta1 and beta2; 
+      1 by default.
+
+  Returns:
+    A 2D Tensor with shape [batch x output_size] equal 
+      to the Multiplicative Integration above.
+
+  Raises:
+    ValueError: if some of the arguments has unspecified or wrong shape.
+  """
+  if args is None or (nest.is_sequence(args) and not args):
+    raise ValueError("`args` must be specified")
+  if not nest.is_sequence(args):
+    raise ValueError("`args` must be list")
+  if len(args) != 2:
+    raise ValueError("`args` must contain 2 tensors")
+  if output_sizes is None or (nest.is_sequence(output_sizes) 
+      and not output_sizes):
+    raise ValueError("`output_sizes` must be specified")
+  if not nest.is_sequence(output_sizes):
+    raise ValueError("`output_sizes` must be list")
+  if not all(isinstance(x, int) for x in output_sizes):
+    raise ValueError("`output_sizes` must contain integers only")
+
+  total_output_size = sum(output_sizes)
+  arg_sizes = []
+  total_arg_size = 0
+  shapes = [x.get_shape() for x in args]
+  for shape in shapes:
+    if shape.ndims != 2:
+      raise ValueError("multiplicative_ntegration is expecting 2D arguments: %s" 
+          % shapes)
+    if shape[1].value is None:
+      raise ValueError("multiplicative_ntegration expects shape[1] to be "
+             "provided for shape %s, but saw %s" % (shape, shape[1]))
+    else:
+      total_arg_size += shape[1].value
+      arg_sizes.append(shape[1].value)
+
+  dtype = [a.dtype for a in args][0]
+
+  scope = vs.get_variable_scope()
+  with vs.variable_scope(scope) as outer_scope:
```

Comment:
you don't need this "with" statement because you're not modifying outer_scope anywhere.

Reply:
the inner_scope seems reasonable; since in it you're disabling the partitioner.  but you don't need the outer_scope because you're not modifying it at all.  so basically just do:

```
with vs.variable_scope(vs.get_variable_scope()) as inner_scope:
   inner_scope.set_partitioner(None)
   ... # set up bias
```

as above, move all this `get_variable` stuff into the `build` methods.

ping me via email (ebrevo@) when ready for re-review.